homeassistant/components/tasmota_irhvac/climate.py

```diff
@@ -0,0 +1,748 @@
+"""Adds support for generic thermostat units."""
+import json
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import mqtt
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    ATTR_PRESET_MODE,
+    ATTR_TEMPERATURE,
+    CONF_NAME,
+    EVENT_HOMEASSISTANT_START,
+    FAN_AUTO,
+    FAN_DIFFUSE,
+    FAN_FOCUS,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_MIDDLE,
+    FAN_OFF,
+    FAN_ON,
+    HVAC_FAN_AUTO,
+    HVAC_FAN_AUTO_MAX,
+    HVAC_FAN_MAX,
+    HVAC_FAN_MAX_HIGH,
+    HVAC_FAN_MEDIUM,
+    HVAC_FAN_MIN,
+    HVAC_MODE_AUTO_FAN,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    HVAC_MODES,
+    PRECISION_HALVES,
+    PRECISION_TENTHS,
+    PRECISION_WHOLE,
+    PRESET_AWAY,
+    PRESET_NONE,
+    STATE_AUTO,
+    STATE_COOL,
+    STATE_DRY,
+    STATE_FAN_ONLY,
+    STATE_HEAT,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_BOTH,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    SWING_VERTICAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+SUPPORT_FLAGS = (
+    SUPPORT_TARGET_TEMPERATURE
+    | SUPPORT_FAN_MODE
+    | SUPPORT_SWING_MODE
+    | SUPPORT_PRESET_MODE
+)
+
+DEFAULT_NAME = "IR AirConditioner"
```

Comment:
I would recommend to move all of these constants to const.py and check if you can reuse any constants from the two constant files I mentioned.

Reply:
Nice suggestion! I was guiding myself by looking how everything is made in other components and I haven't had that idea at all. Thank You for helping me!

Reply:
Of course!

Reply:
Done!